diff --git a/node_modules/@venusprotocol/governance-contracts/contracts/legacy/GovernorAlpha.sol b/node_modules/@venusprotocol/governance-contracts/contracts/legacy/GovernorAlpha.sol
index fca5350..4713034 100644
--- a/node_modules/@venusprotocol/governance-contracts/contracts/legacy/GovernorAlpha.sol
+++ b/node_modules/@venusprotocol/governance-contracts/contracts/legacy/GovernorAlpha.sol
@@ -27,8 +27,8 @@ contract GovernorAlpha {
 
     /// @notice The duration of voting on a proposal, in blocks
     function votingPeriod() public pure returns (uint) {
-        return (60 * 60 * 24 * 3) / 3;
-    } // ~3 days in blocks (assuming 3s blocks)
+        return 100;
+    } // A reasonable amount of block suitable for testing
 
     /// @notice The address of the Venus Protocol Timelock
     TimelockInterface public timelock;
diff --git a/node_modules/@venusprotocol/governance-contracts/contracts/legacy/GovernorAlpha2.sol b/node_modules/@venusprotocol/governance-contracts/contracts/legacy/GovernorAlpha2.sol
index c009718..0e8e8ec 100644
--- a/node_modules/@venusprotocol/governance-contracts/contracts/legacy/GovernorAlpha2.sol
+++ b/node_modules/@venusprotocol/governance-contracts/contracts/legacy/GovernorAlpha2.sol
@@ -27,8 +27,8 @@ contract GovernorAlpha2 {
 
     /// @notice The duration of voting on a proposal, in blocks
     function votingPeriod() public pure returns (uint) {
-        return (60 * 60 * 24 * 3) / 3;
-    } // ~3 days in blocks (assuming 3s blocks)
+        return 100;
+    } // A reasonable amount of block suitable for testing
 
     /// @notice The address of the Venus Protocol Timelock
     TimelockInterface public timelock;
