diff --git a/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/ComptrollerInterface.sol b/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/ComptrollerInterface.sol
index 0981080..854e6e1 100644
--- a/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/ComptrollerInterface.sol
+++ b/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/ComptrollerInterface.sol
@@ -1,6 +1,6 @@
 pragma solidity ^0.5.16;
 
-import "../Tokens/VTokens/VToken.sol";
+import { CorePoolVToken as VToken } from "../Tokens/VTokens/VToken.sol";
 import "../Oracle/PriceOracle.sol";
 import "../Tokens/VAI/VAIControllerInterface.sol";
 import { ComptrollerTypes } from "./ComptrollerStorage.sol";
diff --git a/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/ComptrollerLensInterface.sol b/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/ComptrollerLensInterface.sol
index 2f56b52..fcd8940 100644
--- a/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/ComptrollerLensInterface.sol
+++ b/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/ComptrollerLensInterface.sol
@@ -1,7 +1,7 @@
 pragma solidity ^0.5.16;
 pragma experimental ABIEncoderV2;
 
-import "../Tokens/VTokens/VToken.sol";
+import { CorePoolVToken as VToken } from "../Tokens/VTokens/VToken.sol";
 
 interface ComptrollerLensInterface {
     function liquidateCalculateSeizeTokens(
diff --git a/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/ComptrollerStorage.sol b/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/ComptrollerStorage.sol
index 52af5cc..59d3691 100644
--- a/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/ComptrollerStorage.sol
+++ b/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/ComptrollerStorage.sol
@@ -2,7 +2,7 @@
 
 pragma solidity ^0.5.16;
 
-import { VToken } from "../Tokens/VTokens/VToken.sol";
+import { CorePoolVToken as VToken } from "../Tokens/VTokens/VToken.sol";
 import { PriceOracle } from "../Oracle/PriceOracle.sol";
 import { VAIControllerInterface } from "../Tokens/VAI/VAIControllerInterface.sol";
 import { ComptrollerLensInterface } from "./ComptrollerLensInterface.sol";
diff --git a/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/Diamond/facets/FacetBase.sol b/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/Diamond/facets/FacetBase.sol
index 7d69a6a..d0139d6 100644
--- a/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/Diamond/facets/FacetBase.sol
+++ b/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/Diamond/facets/FacetBase.sol
@@ -2,7 +2,7 @@
 
 pragma solidity 0.5.16;
 
-import { VToken, ComptrollerErrorReporter, ExponentialNoError } from "../../../Tokens/VTokens/VToken.sol";
+import { CorePoolVToken as VToken, ComptrollerErrorReporter, ExponentialNoError } from "../../../Tokens/VTokens/VToken.sol";
 import { IVAIVault } from "../../../Comptroller/ComptrollerInterface.sol";
 import { ComptrollerV16Storage } from "../../../Comptroller/ComptrollerStorage.sol";
 import { IAccessControlManagerV5 } from "@venusprotocol/governance-contracts/contracts/Governance/IAccessControlManagerV5.sol";
diff --git a/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/Diamond/facets/MarketFacet.sol b/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/Diamond/facets/MarketFacet.sol
index e1c9b37..98b78da 100644
--- a/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/Diamond/facets/MarketFacet.sol
+++ b/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/Diamond/facets/MarketFacet.sol
@@ -4,7 +4,7 @@ pragma solidity 0.5.16;
 
 import { IMarketFacet } from "../interfaces/IMarketFacet.sol";
 import { FacetBase } from "./FacetBase.sol";
-import { VToken } from "../../../Tokens/VTokens/VToken.sol";
+import { CorePoolVToken as VToken } from "../../../Tokens/VTokens/VToken.sol";
 
 /**
  * @title MarketFacet
diff --git a/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/Diamond/facets/PolicyFacet.sol b/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/Diamond/facets/PolicyFacet.sol
index 9702f65..5e35254 100644
--- a/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/Diamond/facets/PolicyFacet.sol
+++ b/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/Diamond/facets/PolicyFacet.sol
@@ -2,7 +2,7 @@
 
 pragma solidity 0.5.16;
 
-import { VToken } from "../../../Tokens/VTokens/VToken.sol";
+import { CorePoolVToken as VToken } from "../../../Tokens/VTokens/VToken.sol";
 import { IPolicyFacet } from "../interfaces/IPolicyFacet.sol";
 
 import { XVSRewardsHelper } from "./XVSRewardsHelper.sol";
diff --git a/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/Diamond/facets/RewardFacet.sol b/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/Diamond/facets/RewardFacet.sol
index 641aee0..cb55f7e 100644
--- a/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/Diamond/facets/RewardFacet.sol
+++ b/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/Diamond/facets/RewardFacet.sol
@@ -2,7 +2,7 @@
 
 pragma solidity 0.5.16;
 
-import { VToken } from "../../../Tokens/VTokens/VToken.sol";
+import { CorePoolVToken as VToken } from "../../../Tokens/VTokens/VToken.sol";
 import { IRewardFacet } from "../interfaces/IRewardFacet.sol";
 import { XVSRewardsHelper } from "./XVSRewardsHelper.sol";
 import { SafeBEP20, IBEP20 } from "../../../Utils/SafeBEP20.sol";
diff --git a/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/Diamond/facets/SetterFacet.sol b/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/Diamond/facets/SetterFacet.sol
index d33ac2e..26f425f 100644
--- a/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/Diamond/facets/SetterFacet.sol
+++ b/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/Diamond/facets/SetterFacet.sol
@@ -2,7 +2,7 @@
 
 pragma solidity 0.5.16;
 
-import { VToken } from "../../../Tokens/VTokens/VToken.sol";
+import { CorePoolVToken as VToken } from "../../../Tokens/VTokens/VToken.sol";
 import { ISetterFacet } from "../interfaces/ISetterFacet.sol";
 import { PriceOracle } from "../../../Oracle/PriceOracle.sol";
 import { ComptrollerLensInterface } from "../../ComptrollerLensInterface.sol";
diff --git a/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/Diamond/facets/XVSRewardsHelper.sol b/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/Diamond/facets/XVSRewardsHelper.sol
index 46bb333..15f4ad1 100644
--- a/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/Diamond/facets/XVSRewardsHelper.sol
+++ b/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/Diamond/facets/XVSRewardsHelper.sol
@@ -2,7 +2,7 @@
 
 pragma solidity 0.5.16;
 
-import { VToken } from "../../../Tokens/VTokens/VToken.sol";
+import { CorePoolVToken as VToken } from "../../../Tokens/VTokens/VToken.sol";
 import { FacetBase } from "./FacetBase.sol";
 
 /**
diff --git a/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/Diamond/interfaces/IMarketFacet.sol b/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/Diamond/interfaces/IMarketFacet.sol
index 684d76f..4e8ce87 100644
--- a/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/Diamond/interfaces/IMarketFacet.sol
+++ b/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/Diamond/interfaces/IMarketFacet.sol
@@ -2,7 +2,7 @@
 
 pragma solidity 0.5.16;
 
-import { VToken } from "../../../Tokens/VTokens/VToken.sol";
+import { CorePoolVToken as VToken } from "../../../Tokens/VTokens/VToken.sol";
 
 interface IMarketFacet {
     function isComptroller() external pure returns (bool);
diff --git a/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/Diamond/interfaces/IPolicyFacet.sol b/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/Diamond/interfaces/IPolicyFacet.sol
index 3eac49c..faf85a5 100644
--- a/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/Diamond/interfaces/IPolicyFacet.sol
+++ b/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/Diamond/interfaces/IPolicyFacet.sol
@@ -2,7 +2,7 @@
 
 pragma solidity 0.5.16;
 
-import { VToken } from "../../../Tokens/VTokens/VToken.sol";
+import { CorePoolVToken as VToken } from "../../../Tokens/VTokens/VToken.sol";
 
 interface IPolicyFacet {
     function mintAllowed(address vToken, address minter, uint256 mintAmount) external returns (uint256);
diff --git a/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/Diamond/interfaces/IRewardFacet.sol b/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/Diamond/interfaces/IRewardFacet.sol
index 95b485d..9776762 100644
--- a/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/Diamond/interfaces/IRewardFacet.sol
+++ b/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/Diamond/interfaces/IRewardFacet.sol
@@ -2,7 +2,7 @@
 
 pragma solidity 0.5.16;
 
-import { VToken } from "../../../Tokens/VTokens/VToken.sol";
+import { CorePoolVToken as VToken } from "../../../Tokens/VTokens/VToken.sol";
 import { ComptrollerTypes } from "../../ComptrollerStorage.sol";
 
 interface IRewardFacet {
diff --git a/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/Diamond/interfaces/ISetterFacet.sol b/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/Diamond/interfaces/ISetterFacet.sol
index 0253c88..fac7729 100644
--- a/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/Diamond/interfaces/ISetterFacet.sol
+++ b/node_modules/@venusprotocol/venus-protocol/contracts/Comptroller/Diamond/interfaces/ISetterFacet.sol
@@ -3,7 +3,7 @@
 pragma solidity 0.5.16;
 
 import { PriceOracle } from "../../../Oracle/PriceOracle.sol";
-import { VToken } from "../../../Tokens/VTokens/VToken.sol";
+import { CorePoolVToken as VToken } from "../../../Tokens/VTokens/VToken.sol";
 import { ComptrollerTypes } from "../../ComptrollerStorage.sol";
 import { VAIControllerInterface } from "../../../Tokens/VAI/VAIControllerInterface.sol";
 import { ComptrollerLensInterface } from "../../../Comptroller/ComptrollerLensInterface.sol";
diff --git a/node_modules/@venusprotocol/venus-protocol/contracts/Lens/ComptrollerLens.sol b/node_modules/@venusprotocol/venus-protocol/contracts/Lens/ComptrollerLens.sol
index 47e18e3..2394a23 100644
--- a/node_modules/@venusprotocol/venus-protocol/contracts/Lens/ComptrollerLens.sol
+++ b/node_modules/@venusprotocol/venus-protocol/contracts/Lens/ComptrollerLens.sol
@@ -2,7 +2,7 @@ pragma solidity ^0.5.16;
 pragma experimental ABIEncoderV2;
 
 import "../Tokens/VTokens/VBep20.sol";
-import { VToken } from "../Tokens/VTokens/VToken.sol";
+import { CorePoolVToken as VToken } from "../Tokens/VTokens/VToken.sol";
 import { ExponentialNoError } from "../Utils/ExponentialNoError.sol";
 import "../Tokens/EIP20Interface.sol";
 import "../Oracle/PriceOracle.sol";
diff --git a/node_modules/@venusprotocol/venus-protocol/contracts/Lens/SnapshotLens.sol b/node_modules/@venusprotocol/venus-protocol/contracts/Lens/SnapshotLens.sol
index 11b56be..04303bd 100644
--- a/node_modules/@venusprotocol/venus-protocol/contracts/Lens/SnapshotLens.sol
+++ b/node_modules/@venusprotocol/venus-protocol/contracts/Lens/SnapshotLens.sol
@@ -1,7 +1,7 @@
 pragma solidity ^0.5.16;
 pragma experimental ABIEncoderV2;
 
-import { VToken } from "../Tokens/VTokens/VToken.sol";
+import { CorePoolVToken as VToken } from "../Tokens/VTokens/VToken.sol";
 import { ExponentialNoError } from "../Utils/ExponentialNoError.sol";
 import "../Utils/SafeMath.sol";
 import "../Comptroller/ComptrollerInterface.sol";
diff --git a/node_modules/@venusprotocol/venus-protocol/contracts/Lens/VenusLens.sol b/node_modules/@venusprotocol/venus-protocol/contracts/Lens/VenusLens.sol
index d72fffe..d034667 100644
--- a/node_modules/@venusprotocol/venus-protocol/contracts/Lens/VenusLens.sol
+++ b/node_modules/@venusprotocol/venus-protocol/contracts/Lens/VenusLens.sol
@@ -2,7 +2,7 @@ pragma solidity ^0.5.16;
 pragma experimental ABIEncoderV2;
 
 import "../Tokens/VTokens/VBep20.sol";
-import "../Tokens/VTokens/VToken.sol";
+import { CorePoolVToken as VToken, ExponentialNoError } from "../Tokens/VTokens/VToken.sol";
 import "../Oracle/PriceOracle.sol";
 import "../Tokens/EIP20Interface.sol";
 import "../Tokens/XVS/XVS.sol";
diff --git a/node_modules/@venusprotocol/venus-protocol/contracts/Oracle/PriceOracle.sol b/node_modules/@venusprotocol/venus-protocol/contracts/Oracle/PriceOracle.sol
index 15f44c1..f8f5685 100644
--- a/node_modules/@venusprotocol/venus-protocol/contracts/Oracle/PriceOracle.sol
+++ b/node_modules/@venusprotocol/venus-protocol/contracts/Oracle/PriceOracle.sol
@@ -1,6 +1,6 @@
 pragma solidity ^0.5.16;
 
-import "../Tokens/VTokens/VToken.sol";
+import { CorePoolVToken as VToken } from "../Tokens/VTokens/VToken.sol";
 
 contract PriceOracle {
     /// @notice Indicator that this is a PriceOracle contract (for inspection)
diff --git a/node_modules/@venusprotocol/venus-protocol/contracts/Tokens/VAI/VAIController.sol b/node_modules/@venusprotocol/venus-protocol/contracts/Tokens/VAI/VAIController.sol
index 8b0bb60..49eb857 100644
--- a/node_modules/@venusprotocol/venus-protocol/contracts/Tokens/VAI/VAIController.sol
+++ b/node_modules/@venusprotocol/venus-protocol/contracts/Tokens/VAI/VAIController.sol
@@ -6,7 +6,7 @@ import { VAIControllerErrorReporter } from "../../Utils/ErrorReporter.sol";
 import { Exponential } from "../../Utils/Exponential.sol";
 import { ComptrollerInterface } from "../../Comptroller/ComptrollerInterface.sol";
 import { IAccessControlManagerV5 } from "@venusprotocol/governance-contracts/contracts/Governance/IAccessControlManagerV5.sol";
-import { VToken, EIP20Interface } from "../VTokens/VToken.sol";
+import { CorePoolVToken as VToken, EIP20Interface } from "../VTokens/VToken.sol";
 import { VAIUnitroller, VAIControllerStorageG4 } from "./VAIUnitroller.sol";
 import { VAIControllerInterface } from "./VAIControllerInterface.sol";
 import { VAI } from "./VAI.sol";
diff --git a/node_modules/@venusprotocol/venus-protocol/contracts/Tokens/VTokens/VBNB.sol b/node_modules/@venusprotocol/venus-protocol/contracts/Tokens/VTokens/VBNB.sol
index aa0b516..5352730 100644
--- a/node_modules/@venusprotocol/venus-protocol/contracts/Tokens/VTokens/VBNB.sol
+++ b/node_modules/@venusprotocol/venus-protocol/contracts/Tokens/VTokens/VBNB.sol
@@ -1,6 +1,6 @@
 pragma solidity ^0.5.16;
 
-import "./VToken.sol";
+import { CorePoolVToken as VToken, ComptrollerInterface, InterestRateModel } from "./VToken.sol";
 
 /**
  * @title Venus's vBNB Contract
diff --git a/node_modules/@venusprotocol/venus-protocol/contracts/Tokens/VTokens/VBep20.sol b/node_modules/@venusprotocol/venus-protocol/contracts/Tokens/VTokens/VBep20.sol
index a20a613..7ce7185 100644
--- a/node_modules/@venusprotocol/venus-protocol/contracts/Tokens/VTokens/VBep20.sol
+++ b/node_modules/@venusprotocol/venus-protocol/contracts/Tokens/VTokens/VBep20.sol
@@ -1,6 +1,6 @@
 pragma solidity ^0.5.16;
 
-import { VToken, VBep20Interface, ComptrollerInterface, InterestRateModel, VTokenInterface } from "./VToken.sol";
+import { CorePoolVToken as VToken, VBep20Interface, ComptrollerInterface, InterestRateModel, VTokenInterface } from "./VToken.sol";
 import { EIP20Interface } from "../EIP20Interface.sol";
 import { EIP20NonStandardInterface } from "../EIP20NonStandardInterface.sol";
 
diff --git a/node_modules/@venusprotocol/venus-protocol/contracts/Tokens/VTokens/VToken.sol b/node_modules/@venusprotocol/venus-protocol/contracts/Tokens/VTokens/VToken.sol
index 7dd98f1..964ae46 100644
--- a/node_modules/@venusprotocol/venus-protocol/contracts/Tokens/VTokens/VToken.sol
+++ b/node_modules/@venusprotocol/venus-protocol/contracts/Tokens/VTokens/VToken.sol
@@ -14,7 +14,7 @@ import { IAccessControlManagerV5 } from "@venusprotocol/governance-contracts/con
  * @notice Abstract base for vTokens
  * @author Venus
  */
-contract VToken is VTokenInterface, Exponential, TokenErrorReporter {
+contract CorePoolVToken is VTokenInterface, Exponential, TokenErrorReporter {
     struct MintLocalVars {
         MathError mathErr;
         uint exchangeRateMantissa;
diff --git a/node_modules/@venusprotocol/venus-protocol/contracts/test/MockVBNB.sol b/node_modules/@venusprotocol/venus-protocol/contracts/test/MockVBNB.sol
index adf9c4c..e8dcfb3 100644
--- a/node_modules/@venusprotocol/venus-protocol/contracts/test/MockVBNB.sol
+++ b/node_modules/@venusprotocol/venus-protocol/contracts/test/MockVBNB.sol
@@ -1,6 +1,6 @@
 pragma solidity ^0.5.16;
 
-import "../Tokens/VTokens/VToken.sol";
+import { CorePoolVToken as VToken, ComptrollerInterface, InterestRateModel } from "../Tokens/VTokens/VToken.sol";
 
 /**
  * @title Venus's vBNB Contract
diff --git a/node_modules/@venusprotocol/venus-protocol/contracts/test/VBep20MockDelegate.sol b/node_modules/@venusprotocol/venus-protocol/contracts/test/VBep20MockDelegate.sol
index 4108562..279d1cb 100644
--- a/node_modules/@venusprotocol/venus-protocol/contracts/test/VBep20MockDelegate.sol
+++ b/node_modules/@venusprotocol/venus-protocol/contracts/test/VBep20MockDelegate.sol
@@ -1,6 +1,6 @@
 pragma solidity ^0.5.16;
 
-import "../Tokens/VTokens/VToken.sol";
+import { CorePoolVToken as VToken, ComptrollerInterface, InterestRateModel, VBep20Interface, VTokenInterface, EIP20NonStandardInterface, EIP20Interface } from "../Tokens/VTokens/VToken.sol";
 
 /**
  * @title Venus's VBep20 Contract
